Tests/SonarScanner.MSBuild.Tasks.IntegrationTests/E2ETests/E2EAnalysisTests.cs

```diff
@@ -254,124 +267,199 @@ public void E2E_HasManagedAndContentFiles_VB()
 
             // Add files under the obj folder that should not be analyzed
-            var objFolder = Path.Combine(projectFolder, "obj");
+            var objFolder = Path.Combine(rootInputFolder, "obj");
             var objSubFolder1 = Path.Combine(objFolder, "debug");
             var objSubFolder2 = Path.Combine(objFolder, "xxx"); // any folder under obj should be ignored
             Directory.CreateDirectory(objSubFolder1);
             Directory.CreateDirectory(objSubFolder2);
 
             // File in obj
-            var filePath = CreateEmptyFile(objFolder, "cs");
-            descriptor.AddCompileInputFile(filePath, false);
+            var objFile = CreateEmptyFile(objFolder, "objFile1.cs");
 
             // File in obj\debug
-            filePath = CreateEmptyFile(objSubFolder1, "cs");
-            descriptor.AddCompileInputFile(filePath, false);
+            var objDebugFile = CreateEmptyFile(objSubFolder1, "objDebugFile1.cs");
 
             // File in obj\xxx
-            filePath = CreateEmptyFile(objSubFolder2, "vb");
-            descriptor.AddCompileInputFile("obj\\xxx\\" + Path.GetFileName(filePath), false); // reference the file using a relative path
-
-            var preImportProperties = CreateDefaultAnalysisProperties(rootInputFolder, rootOutputFolder);
+            var objFooFile = CreateEmptyFile(objSubFolder2, "objFooFile.cs");
+
+            var projectXml = $@"
+<ItemGroup>
+  <Compile Include='{compile1}' />
+  <Compile Include='foo\compile2.cs' />
+  <Compile Include='{objFile}' />
+  <Compile Include='obj\debug\objDebugFile1.cs' />
+  <Compile Include='{objFooFile}' />
+</ItemGroup>
+";
+            var projectFilePath = CreateProjectFile(projectXml, rootOutputFolder, isVB: true);
 
             // Act
-            var projectDir = CreateAndBuildSonarProject(descriptor, rootOutputFolder, preImportProperties);
+            var result = BuildRunner.BuildTargets(TestContext, projectFilePath);
 
-            AssertFileExists(projectDir, ExpectedAnalysisFilesListFileName);
+            // Assert
+            result.AssertTargetSucceeded(TargetConstants.DefaultBuildTarget); // Build should succeed with warnings
+            var projectSpecificOutputDir = CheckProjectSpecificOutputStructure(rootOutputFolder);
 
-            CheckProjectOutputFolder(descriptor, projectDir);
+            // Check the list of files to be analyzed
+            var expectedFilesToAnalyzeFilePath = AssertFileExists(projectSpecificOutputDir, ExpectedAnalysisFilesListFileName);
+            var fileList = File.ReadLines(expectedFilesToAnalyzeFilePath);
+            fileList.Should().BeEquivalentTo(new string[]
+            {
+                rootInputFolder + "\\compile1.cs",
+                rootInputFolder + "\\foo\\compile2.cs"
+            });
         }
 
         [TestCategory("E2E"), TestCategory("Targets")] // SONARMSBRU-12: Analysis build fails if the build definition name contains brackets
+        [TestMethod]
```

Comment:
Added the missing `TestMethod` attribute so this test actually executes.

As it happens, the old test failed when I added the attribute, as it wasn't copying all of the required files. I fixed the old test, then converted it to the new format.